R/assess_reverse_dependencies.R

```diff
@@ -0,0 +1,48 @@
+#' Generate list of Reverse Dependencies for a package
+#'
+#' @details The more packages that depend on a package the more chance
+#' for errors/bugs to be found
+#'
+#' @eval roxygen_assess_family(
+#'   "reverse_dependencies",
+#'   "A character vector of reverse dependencies")
+#'
+#' @export
+assess_reverse_dependencies <- function(x, ...){
+  UseMethod("assess_reverse_dependencies")
+}
+
+attr(assess_remote_checks, "column_name") <- "reverse_dependencies"
+attributes(assess_remote_checks)$label <- "List of reverse dependencies a package has"
+
+#' @importFrom devtools revdep
+#' @export
+assess_reverse_dependencies.default <- function(x, ...){
+  pkg_metric_eval(class = "pkg_metric_reverse_dependencies",
+                  devtools::revdep(x$name, bioconductor = TRUE)
+  )
+}
+
+#' @importFrom devtools revdep
+#' @export
+assess_reverse_dependencies.cohort_ref <- function(x, ...){
+  #cohort_rev_deps <- devtools::revdep(x$cohort$name, bioconductor = TRUE)
+  cohort_rev_deps <- lapply(x$cohort, assess_reverse_dependencies)
```

Comment:
Super weird S3 dispatch within an apply - try this:

```suggestion
  cohort_rev_deps <- lapply(x$cohort, function(i) assess_reverse_dependencies(i))
```

Reply:
```
cohort_rev_deps <-x$cohort %>% assess_reverse_dependencies()

```
or

```
cohort_rev_deps <-x$cohort %>% pkg_assess( assessments = assess_reverse_dependencies())
```

Reply:
I don't think we ever want to use `pkg_assess` on a cohort directly. Instead, maybe we need a `cohort_assess` helper that applies all available package assessments (likely using `pkg_assess` under the hood), but also including context and cohort metrics. 

Let's take two examples, a generic "context" assessment `assess_context_metric` and a generic "cohort" assessment `assess_cohort_metric`. Just to recap (and make sure I'm remembering correctly), a context metric is a per-package metric that is contextualized by a cohort, and a cohort metric is a per-cohort metric (producing a single score for an entire cohort). 

Then we'd have these scenarios

### Given a cohort, calculate a specific _context_ metric for _a single_ package

```
# just mocking up some example params, feel free to suggest alternatives
assess_reverse_dependencies(<pkg>, context = <cohort>, ...) -> <cohort_of_pkg_metric (len 1)>
```

### Given a cohort, calculate a specific _context_ metric for _all_ cohort packages

```
assess_reverse_dependencies(<cohort>, ...) -> <cohort_of_pkg_metric (len n)>
```

### Given a cohort, calculate a single _cohort_ metric

```
assess_cohort_metric(<cohort>, ...) -> <cohort_metric>
```

### Given a cohort, calculate all _context_ & _cohort_ metrics for _all_ cohort packages

Analogous to how `pkg_assess` is a convenience function for calculating all available assessments for a `list_of_pkg_ref` objects, we need a way to run all the available cohort-style metrics on a cohort. The return type should be similar to `pkg_assess` (one metric per assessment per package), but with additional cohort metrics (one metric per cohort metric)

```
cohort_assess(<cohort>, ...)
```